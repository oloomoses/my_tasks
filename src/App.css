/* RESETS */
*,
*::before,
*::after {
  box-sizing: border-box;
}
*:focus {
  outline: 3px dashed #228bec;
  outline-offset: 0;
}
/* html {
  font: 62.5% / 1.15 sans-serif;
} */
h1,
h2 {
  margin-bottom: 0;
}
ul {
  list-style: none;
  padding: 0;
}
/* button {
  border: none;
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;
  background: transparent;
  color: inherit;
  font: inherit;
  line-height: normal;
  -webkit-font-smoothing: inherit;
  -moz-osx-font-smoothing: inherit;
  -webkit-appearance: none;
} */
/* button::-moz-focus-inner {
  border: 0;
} */
/* button,
input,
optgroup,
select,
textarea {
  font-family: inherit;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
} */
/* button,
input {
  overflow: visible;
}
input[type="text"] {
  border-radius: 0;
} */

body {
  margin: 0;
  padding: 0;
  font-family:'Courier New', Courier, monospace;  
}

.light {
  background-color: #f5f5f5;
  color: #4d4d4d;
}

.dark {
  background-color: #000000;
  color: #fff;
}
@media screen and (min-width: 620px) {
  body {
    font-size: 1.9rem;
    line-height: 1.31579;
    height: auto;
  }
}
/* END RESET */

.App {
  width: 100%;
  max-width: 58rem;
  margin: 0 auto;
}
.container {
  margin-left: 2em;
  margin-right: 2em;
}

.nav {  
  display: grid;
  grid-template-columns: 2fr 1fr;
  margin-top: 2rem;
}

.icons {
  justify-self: end;
  display: flex;
  grid-template-columns: 1fr 1fr;
  gap: 2em;
}

.form-title,
.header-title {
  font-size: x-large;
  font-weight: 500;
  margin-bottom: 0.5em;
}

.task-list {
  margin-top: 1em;
}

.task {
  display: grid;
  grid-template-columns: 20% 60% 20%;
  margin-top: 1em;
}

.avatar {
  /* margin-right: 2em; */
}

.task-td {
  opacity: 0.7;
  justify-self: start;
}

.completed-time,
.task-description {
  font-size: x-small;
  opacity: 0.7;
}

.cb {
  justify-self: self-end;
}

.stats-title {
  font-size: large;
  margin-top: 2em;
  opacity: 0.9;
}

.stat {
  display: grid;
  margin-top: 2em;
  grid-template-columns: 1fr 3fr;
  align-items: end;
}

.percentage-description, .stat-description {
  font-size: small;
}

.stat-description {
  color: #F17031;
}

.stat-value {
  font-size: large;
  opacity: 0.7;
}

hr {
  margin-top: 2em;
  opacity: 0.2;
}

.header-background {
  background-color: #F17031;
  width: 100%;
  height: 3em;
}

/* FORM */

form {
  display: flex;
  flex-direction: column;
}

.btn,
.task-input {
  width: 100%;
  height: 2.5rem;
  border: 0.5px solid #a7a4a4;
  border-radius: 5px;
  margin-top: 1rem;
  padding: 0.5rem;
  font-size: 0.9rem;  
}

.btn {
  width: 5rem;
  margin-right: auto;
  margin-left: auto;
  background-color: #F17031;
  color: #fff;
  font-weight: bold;
  border: transparent;
}
