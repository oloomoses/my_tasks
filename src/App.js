import './App.css';
import { useState } from 'react';
import { Routes, Route } from 'react-router-dom';
import { nanoid } from 'nanoid';
import Nav from './components/Nav';
import TaskList from './components/TaskList';
import TaskAdd from './components/TaskAdd';
import NotFound from './components/NotFound';

const DATA = [
  {
    id: 1, name: 'Task name 1', description: 'task name 1 description', avatar: '', complete: false, completedTime: '',
  },
  {
    id: 2, name: 'Task name 2', description: 'task name 2 description', avatar: '', complete: false, completedTime: '',
  },
  {
    id: 3, name: 'Task name 3', description: 'task name 3 description', avatar: '', complete: false, completedTime: '',
  },
];

function App() {
  const [mode, setMode] = useState('light');
  const [tasks, setTasks] = useState(DATA);

  const modeToggle = () => (mode === 'light' ? setMode('dark') : setMode('light'));

  const addTask = (name, description, avatar) => {
    const newTask = {
      id: `id-${nanoid()}`, name, description, avatar, complete: false, completedTime: '',
    };
    setTasks([...tasks, newTask]);
  };

  const updateTask = (updatedTask) => {
    setTasks(updatedTask);
  };

  return (
    <div className="">
      <div className="header-background" />
      <div className="App">
        <div className="container">
          <Nav toggle={modeToggle} mode={mode} />
          <Routes>
            <Route path="/" element={<TaskList tasks={tasks} updateTask={updateTask} />} />
            <Route path="/addTask" element={<TaskAdd addTask={addTask} />} />
            <Route path="*" element={<NotFound />} />
          </Routes>

        </div>
      </div>
    </div>
  );
}

export default App;
