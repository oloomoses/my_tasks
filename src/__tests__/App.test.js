import { shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import App from '../App';

let wrapper;
beforeEach(() => {
  wrapper = shallow(<BrowserRouter><App /></BrowserRouter>);
});

test('renders Nav component', () => {
  expect(wrapper.find('Nav')).toBeTruthy();
});

test('renders TaskList component', () => {
  expect(wrapper.find('TaskList')).toBeTruthy();
});

test('renders AddTask component', () => {
  expect(wrapper.find('AddTask')).toBeTruthy();
});

test('renders NotFound component', () => {
  expect(wrapper.find('NotFound')).toBeTruthy();
});
