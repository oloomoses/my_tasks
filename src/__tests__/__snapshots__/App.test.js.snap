// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className=""
>
  <div
    className="header-background"
  />
  <div
    className="App"
  >
    <div
      className="container"
    >
      <div
        className="nav"
      >
        <div
          className="header-title"
        >
          Tasks
        </div>
        <div
          className="icons"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium night-mode css-i4bv87-MuiSvgIcon-root"
            data-testid="Brightness4SharpIcon"
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6z"
            />
          </svg>
          <a
            href="/addTask"
            onClick={[Function]}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium add-icon css-i4bv87-MuiSvgIcon-root"
              data-testid="AddIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </a>
        </div>
      </div>
      <ul
        className="task-list"
      >
        <li
          className="task stack-small"
        >
          <div
            className="avatar"
          >
            <img
              alt="A"
              className="avatar-img"
              onError={[Function]}
              src=""
            />
          </div>
          <div
            className="task-td"
          >
            <div
              className="task-title"
            >
              Workout
            </div>
            <div
              className="task-description"
            >
              Weight lifting
            </div>
          </div>
          <div
            className="cb"
          >
            <input
              defaultChecked={false}
              id="todo-0"
              onChange={[Function]}
              type="checkbox"
            />
          </div>
        </li>
        <li
          className="task stack-small"
        >
          <div
            className="avatar"
          >
            <img
              alt="A"
              className="avatar-img"
              onError={[Function]}
              src=""
            />
          </div>
          <div
            className="task-td"
          >
            <div
              className="task-title"
            >
              Shopping
            </div>
            <div
              className="task-description"
            >
              Buy groceries
            </div>
          </div>
          <div
            className="cb"
          >
            <input
              defaultChecked={false}
              id="todo-0"
              onChange={[Function]}
              type="checkbox"
            />
          </div>
        </li>
        <li
          className="task stack-small"
        >
          <div
            className="avatar"
          >
            <img
              alt="A"
              className="avatar-img"
              onError={[Function]}
              src=""
            />
          </div>
          <div
            className="task-td"
          >
            <div
              className="task-title"
            >
              Study
            </div>
            <div
              className="task-description"
            >
              Codding challenges
            </div>
          </div>
          <div
            className="cb"
          >
            <input
              defaultChecked={false}
              id="todo-0"
              onChange={[Function]}
              type="checkbox"
            />
          </div>
        </li>
      </ul>
      <hr />
      <div
        className="stats-title"
      >
        Stats
      </div>
      <ul
        className="stats"
      >
        <li
          className="stat stack-small"
        >
          <div
            className="stat-value"
          >
            0
          </div>
          <div
            className="stat-description"
          >
            Total Tasks Closed
          </div>
        </li>
        <li
          className="stat stack-small"
        >
          <div
            className="stat-value"
          >
            3
          </div>
          <div
            className="stat-description"
          >
            Total Tasks Open
          </div>
        </li>
        <li
          className="stat stack-small"
        >
          <div
            className="stat-value"
          >
            0
            %
          </div>
          <div
            className="percentage-description"
          >
            Completion Rate
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
